yaksh/models.py

```diff
@@ -134,6 +140,346 @@ def create_trial_course(self, user):
+                trial_quiz.attempts_allowed = -1
+                trial_quiz.active = True
+                trial_quiz.start_date_time = timezone.now()
+                trial_quiz.end_date_time = datetime(
+                    2199, 1, 1, 0, 0, 0, 0, tzinfo=pytz.utc
+                )
+            trial_quiz.save()
+
+        # Get or create Trial Ordered Lesson for usermode/godmode
+        learning_modules = trial_course.get_learning_modules()
+        if learning_modules:
+            quiz = learning_modules[0].learning_unit.filter(quiz=trial_quiz)
+            if not quiz.exists():
+                trial_learning_unit = LearningUnit.objects.create(
+                    order=1, quiz=trial_quiz, learning_type="quiz",
+                    check_prerequisite=False)
+                learning_modules[0].learning_unit.add(trial_learning_unit.id)
+            trial_learning_module = learning_modules[0]
+        else:
+            trial_learning_module = LearningModule.objects.create(
+                name="Trial for {}".format(trial_course), order=1,
+                check_prerequisite=False, creator=user, is_trial=True)
+            trial_learning_unit = LearningUnit.objects.create(
+                order=1, quiz=trial_quiz, learning_type="quiz",
+                check_prerequisite=False)
+            trial_learning_module.learning_unit.add(trial_learning_unit.id)
+            trial_course.learning_module.add(trial_learning_module.id)
+
+        # Add user to trial_course
+        trial_course.enroll(False, user)
+        return trial_quiz, trial_course, trial_learning_module
+
+
+###############################################################################
+class Quiz(models.Model):
+    """A quiz that students will participate in. One can think of this
+    as the "examination" event.
+    """
+
+    # The start date of the quiz.
+    start_date_time = models.DateTimeField(
+        "Start Date and Time of the quiz",
+        default=timezone.now,
+        null=True
+    )
+
+    # The end date and time of the quiz
+    end_date_time = models.DateTimeField(
+        "End Date and Time of the quiz",
+        default=datetime(
+            2199, 1, 1,
+            tzinfo=pytz.timezone(timezone.get_current_timezone_name())
+        ),
+        null=True
+    )
+
+    # This is always in minutes.
+    duration = models.IntegerField("Duration of quiz in minutes", default=20)
+
+    # Is the quiz active. The admin should deactivate the quiz once it is
+    # complete.
+    active = models.BooleanField(default=True)
+
+    # Description of quiz.
+    description = models.CharField(max_length=256)
+
+    # Mininum passing percentage condition.
+    pass_criteria = models.FloatField("Passing percentage", default=40)
+
+    # Number of attempts for the quiz
+    attempts_allowed = models.IntegerField(default=1, choices=attempts)
+
+    time_between_attempts = models.IntegerField(
+        "Number of Days", choices=days_between_attempts
```

Comment:
This is always hard to follow on the UI, change "Number of Days" to "Days between attempts", should this be the number of hours instead of days?

Reply:
Will be changed in a separate PR.